### DISCUSSION TITLE
New Model/Code: Segment Everything Everywhere All at Once

### DISCUSSION THREAD

**Discussion by JaredS215** [continue-revolution/sd-webui-segment-anything#71] [General]
Looks interesting!
https://github.com/UX-Decoder/Segment-Everything-Everywhere-All-At-Once

**Comment by continue-revolution** [comment]
I just took a look, but it seems that all features except video features (not planning to support video) have been supported in this extension. Please correct me if you find other interesting features they have but I do not have, and I’m glad to support.

https://github.com/UX-Decoder/Segment-Everything-Everywhere-All-At-Once#camping-click-scribble-to-mask This should have been supported in the earliest version of this extension. Scribble is actually click multiple times (this has to do with SAM’s API)

https://github.com/UX-Decoder/Segment-Everything-Everywhere-All-At-Once#mountain_snow-text-to-mask This should be
achieved by using GroundingDINO or Oneformer semantic segmentation from ControlNet extension. However, a probable limitation is that click/text and oneformer are separated in two panels, this is because oneformer cannot operate on arbitrary size, and I want to avoid any kind of unnecessary resize.

https://github.com/UX-Decoder/Segment-Everything-Everywhere-All-At-Once#mosque-referring-image-to-mask This is probably to get the categoty/text expression from the input image, and use this text/category expression to generate segmentation.
